WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -109,87 +113,23 @@ private List<AliceClient> CreateAliceClients(RoundState roundState)
 			return aliceClients;
 		}
 
-		private async Task<List<AliceClient>> RegisterCoinsAsync(IEnumerable<AliceClient> aliceClients, CancellationToken cancellationToken)
+		private async Task RegisterCoinsAsync(IEnumerable<AliceClient> aliceClients, CancellationToken cancellationToken)
 		{
-			async Task<AliceClient?> RegisterInputTask(AliceClient aliceClient)
+			async Task RegisterInputTask(AliceClient aliceClient)
 			{
-				try
-				{
-					await aliceClient.RegisterInputAsync(cancellationToken).ConfigureAwait(false);
-					return aliceClient;
-				}
-				catch (Exception e)
-				{
-					Logger.LogWarning($"Round ({aliceClient.RoundId}), Alice ({aliceClient.AliceId}): {nameof(AliceClient.RegisterInputAsync)} failed, reason:'{e}'.");
-					return default;
-				}
+				await aliceClient.RegisterInputAsync(cancellationToken).ConfigureAwait(false);
 			}
 
 			var registerRequests = aliceClients.Select(RegisterInputTask);
-			var completedRequests = await Task.WhenAll(registerRequests).ConfigureAwait(false);
-
-			return completedRequests.Where(x => x is not null).Cast<AliceClient>().ToList();
-		}
-
-		private async Task<List<AliceClient>> ConfirmConnectionsAsync(IEnumerable<AliceClient> aliceClients, long maxVsizeAllocationPerAlice, TimeSpan connectionConfirmationTimeout, CancellationToken cancellationToken)
-		{
-			async Task<AliceClient?> ConfirmConnectionTask(AliceClient aliceClient)
-			{
-				try
-				{
-					await aliceClient.ConfirmConnectionAsync(connectionConfirmationTimeout, maxVsizeAllocationPerAlice, cancellationToken).ConfigureAwait(false);
-					return aliceClient;
-				}
-				catch (Exception e)
-				{
-					Logger.LogWarning($"Round ({aliceClient.RoundId}), Alice ({aliceClient.AliceId}): {nameof(AliceClient.ConfirmConnectionAsync)} failed, reason:'{e}'.");
-					return default;
-				}
-			}
-
-			var confirmationRequests = aliceClients.Select(ConfirmConnectionTask);
-			var completedRequests = await Task.WhenAll(confirmationRequests).ConfigureAwait(false);
-
-			return completedRequests.Where(x => x is not null).Cast<AliceClient>().ToList();
+			await Task.WhenAll(registerRequests).ConfigureAwait(false);
 		}
 
-		private IEnumerable<Money> DecomposeAmounts(FeeRate feeRate)
+		private IEnumerable<Money> DecomposeAmounts(FeeRate feeRate, Money minimumOutputAmount, IEnumerable<Money>? forcedOutputDenominations = null)
```

Comment:
TBH we don't need this feature, it should only be a constructor argument. The reason I put it in an object instead of making the whole thing a function is that for weighted random selection of denominations in the jupyter notebook I precompute cumulative weight values, since decomposition is used thousands of times in the randomized strategies.

Reply:
The current tests are relying on this parameter. I think this could be removed when we will add the new decomposer. @nothingmuch can you add this?

Reply:
mentioned in #5946